src/components/SubRedditForm.js

```diff
@@ -21,10 +32,61 @@ export default function SubRedditForm() {
     e.preventDefault();
     history.push(`/search/${input}`);
   };
+
   useEffect(() => {
     setInput(() => `${subreddit}`);
-  }, [subreddit]);
+    // clean up after
+    return () => {
+      setInput(() => '');
+    };
+  }, [fullData, subreddit]);
+
+  function axiosGet(sreddit, after) {
+    return axios.get(
+      `https://www.reddit.com/r/${sreddit}/top.json?t=year&limit=100${after ? `&after=${after}` : ''}`
+    );
+  }
+
+  // load data area
+  useEffect(() => {
+    async function fetchData() {
+      setLoading(() => true);
+      const fetchAllResults = async () => {
+        let after = null;
+        let dataStore = [];
+        for (let i = 0; i < 5; i += 1) {
+          // eslint-disable-next-line no-await-in-loop
+          const results = await axiosGet(subreddit, after);
+
+          // console.log(results.data.data);
+          // console.log(results.data.data.children);
+
+          dataStore = [...dataStore, ...results.data.data.children];
+          after = results.data.after;
+        }
+
+        return dataStore;
+      };
```

Comment:
You were almost there :smile: Let me show you my debugging process.

First I looked at the code. It seems like all the parts are there at first glance.

Second I ran the app and checked the network requests. This is what I saw:

![image](https://user-images.githubusercontent.com/9722792/98436724-098d0f00-20de-11eb-9e22-20b838fb9782.png)

Here you can see a couple of things:

1. There are five requests that are sent one after another and not in parallel (you can see that thanks to the green bars on the right). So this works fine. The requests need to be sent one after another
2. All requests are the same. None of them include the `after` parameter. This is not correct since it will give us five times the same response.

Now the question is: Why is the after param not included in the URL? You seem to set it correctly in the `axiosGet` function.

Next, we can check what the after param looks like for each iteration of the for loop. So I added a simple `console.log` statement before `axiosGet`.

```
for (let i = 0; i < 5; i += 1) {
  console.log('start fetch', subreddit, after)
  // eslint-disable-next-line no-await-in-loop
  const results = await axiosGet(subreddit, after);
```

Now when this runs I see the following output in my console.

![image](https://user-images.githubusercontent.com/9722792/98436823-ddbe5900-20de-11eb-83ef-58d72a736b9e.png)

So it seems like the `after` variable is `undefined` in each iteration. That's strange because you seem to update it in every iteration in the following line

```
after = results.data.after;
```

So how come `after` is always undefined?

You correctly added a couple of `console.log` statements at some point

```
// console.log(results.data.data);
// console.log(results.data.data.children);
```

so you were clearly on the right path. Now let's see what happens when we add this log statement

```
const results = await axiosGet(subreddit, after);
console.log('results', results)
console.log('results.data.after', results.data.after);
console.log('results.data.data.children', results.data.data.children);
```

You can see this output in the console

![image](https://user-images.githubusercontent.com/9722792/98437014-748b1580-20df-11eb-9527-000691484cff.png)

So it seems that `results.data.after` is `undefined` which explains why the `after` variable outside the for loop is always undefined.

Now when you have a closer look at the output of `results` you can see this 

![image](https://user-images.githubusercontent.com/9722792/98437057-b916b100-20df-11eb-9493-842c2a1bb142.png)

The `after` field is set. But it's nested deeper than `results.data.after`. It has to be 

```
results.data.data.after
```

Let's update the corresponding `console.log` statement to

```
console.log('results.data.data.after', results.data.data.after);
```

and the line where you set the `after` variable

```
after = results.data.data.after;
```

Now when you look at the console output you should see

![image](https://user-images.githubusercontent.com/9722792/98437115-1c084800-20e0-11eb-98ca-47ab66601284.png)

Which is great. Now check the network requests

![image](https://user-images.githubusercontent.com/9722792/98437121-2d515480-20e0-11eb-8249-ca685896eb1f.png)

And it seems to work. 5 requests with different `after` params.